### ISSUE TITLE
ob deploy commands steal the foreground / behave strangely

### ISSUE THREAD

**Issue by ORESoftware** [obsidiansystems/obelisk#501]
I have this:

```
        (
          ob deploy init --ssh-key ~/interos-prod.pem  --hostname "$server" --route "$route" --admin-email $email "$deploy_dir"
        ) &

        wait;

        cd "$deploy_dir";
        
       # do some more stuff .. etc etc

        echo 'OK here we go, ob deploy pushing....';

        (
           ob deploy push -v
        ) &
        
        deploy_pid="$!"
        
        wait "$deploy_pid" || {
         echo 'deploying with ob deploy push exited with non zero'
         exit 1;
        }
```

what happens is that my script will stop at any `ob deploy` command unless I put them in the background...it's possibly that it's the same behavior as `exec foo` in the shell - it steals the foreground and will exit the current shell without whatever the exit code is of `exec foo`...seems to be the same behavior as `ob deploy init` and `ob deploy push`

**Comment by vaibhavsagar** [comment]
Why are you running `ob deploy init` repeatedly? It's only meant to be run once, like `git init`.

**Comment by ORESoftware** [comment]
@vaibhavsagar don't worry about that right now, would like to understand/tackle OP, because `ob deploy push` behaves the same way in this regard

**Comment by vaibhavsagar** [comment]
If you don't have to run `ob deploy init` multiple times, then you don't need the rest of your script because you can instead deploy directly:

```bash
$ ob deploy push -v
```

so I don't see what the problem is.

**Comment by ORESoftware** [comment]
it's the same problem when you create a script like this:

```
#!/bin/bash

echo 'foo'
exec echo 'foo'
echo 'this is bar'

```

you will notice that 'this is bar' does not get echoed. The same behavior occurs with `ob` commands. interesting isn't it.

**Comment by 3noch** [comment]
Running `exec echo 'foo'` in a terminal will close it. `ob` commands don't behave that way.

**Comment by ryantrinkle** [comment]
@ORESoftware Is it possible [`-e` is set](https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html) and the commands are failing?  Also, would you mind posting the script that exhibits the problem so we can compare it to the one with the fixes in place?

**Comment by ORESoftware** [comment]
yes, I have `set -e` set, how does that effect `ob deploy` commands?

**Comment by ryantrinkle** [comment]
With `set -e`, your script will bail out if (almost) any command fails
(i.e. returns a nonzero exit code). Could that be what's happening here?

**Comment by ORESoftware** [comment]
it's not what's happening as far as I can tell, it behaves just like the exec command, it exits, and the parent script exits with it, even with `ob deploy` exiting with 0

**Comment by ORESoftware** [comment]
it's not a big deal, I can solve it using:

```
  (
    ob deploy init --ssh-key ~/interos-prod.pem  --hostname "$server" --route "$route" --admin-email $email "$deploy_dir"
  ) &

```

but it's just something you might want to know about

**Comment by 3noch** [comment]
@ORESoftware If you use `-v` on your `ob` commands, does it fix the issue?
